test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
+            fmt = '%B',
+        },
+        {
+            buf = 'December',
+            fmt = '%B',
+        },
+        -- %c
+        {
+            buf = 'Thu Jan  1 03:00:00 1970',
+            fmt = '%c',
+        },
+        -- %C
+        {
+            buf = '26-08-2024 20',
+            fmt = '%d-%m-%Y %C',
+        },
+        -- %d
+        {
+            buf = '01',
+            fmt = '%d',
+        },
+        {
+            buf = '31',
+            fmt = '%d',
+        },
+        -- %D
+        {
+            buf = '09/09/04',
+            fmt = '%D',
+        },
+        -- %e
+        {
+            buf = ' 30',
+            fmt = '%e',
+        },
+        -- %E, modifier, skipped.
+        -- %F
+        {
+            buf = '1970-01-01T03:00:00.125',
+            fmt = '%FT%T.%f',
+        },
+        {
+            buf = '1970-01-01T03:00:00.125 MSK',
+            fmt = '%FT%T.%f %Z',
+        },
+        {
+            buf = '1970-01-01T03:00:00.125+0300',
+            fmt = '%FT%T.%f%z',
+        },
+        {
+            buf = '1970-01-01',
+            fmt = '%F',
+        },
+        -- %f, Tarantool-specific.
+        {
+            buf = '100',
+            fmt = '%f',
+        },
+        -- %G
+        {
+            buf = '2024',
+            fmt = '%G',
+        },
+        -- %g, it is not possible to roundtrip, skipped.
+        -- %h, same as %b
+        {
+            buf = 'Jan',
+            fmt = '%b',
+        },
+        {
+            buf = 'Feb',
+            fmt = '%b',
+        },
+        {
+            buf = 'Mar',
+            fmt = '%b',
+        },
+        {
+            buf = 'Apr',
+            fmt = '%b',
+        },
+        {
+            buf = 'May',
+            fmt = '%b',
+        },
+        {
+            buf = 'Jun',
+            fmt = '%b',
+        },
+        {
+            buf = 'Jul',
+            fmt = '%b',
+        },
+        {
+            buf = 'Aug',
+            fmt = '%b',
+        },
+        {
+            buf = 'Sep',
+            fmt = '%b',
+        },
+        {
+            buf = 'Oct',
+            fmt = '%b',
+        },
+        {
+            buf = 'Nov',
+            fmt = '%b',
+        },
+        {
+            buf = 'Dec',
+            fmt = '%b',
+        },
+        -- %H
+        {
+            buf = '00',
+            fmt = '%H',
+        },
+        {
+            buf = '23',
+            fmt = '%H',
+        },
+        -- %I
+        {
+            buf = '01',
+            fmt = '%I',
+        },
+        {
+            buf = '12',
+            fmt = '%I',
+        },
+        -- %j
+        {
+            buf = '001',
+            fmt = '%j',
+        },
+        {
+            buf = '31-12-2024 366',
+            fmt = '%d-%m-%Y %j',
+        },
+        -- %k
+        {
+            buf = ' 0',
+            fmt = '%k',
+        },
+        {
+            buf = '23',
+            fmt = '%k',
+        },
+        -- %l
```

Comment:
Looks like it is synonym `%I`, should we double the tests here?

Reply:
The proposed testcases tests datetime by specification, not by code. These tests guarantee that all specification conversions described in a POSIX standard works as expected. So the answer is "yes".